packages/backend/helpers/rateLimitMiddleware.ts

```diff
@@ -0,0 +1,50 @@
+import { Request, Response } from 'express';
+import { RateLimiterRedis, IRateLimiterStoreOptions } from 'rate-limiter-flexible';
+
+import redis from '../redis/client';
+import { skipRateLimitRoutes } from './utils';
+import config from '../config';
+
+// In Memory Cache for storing RateLimiterRedis instances to prevent the creation of a new instance for each request.
+// The cache key is the 'rate_limit' value derived from the subscriptions table. Currently, we cache by the 'rate_limit' value for simplicity.
+// Using 'subscriptionId' as a key would be more precise but would add complexity in keeping the cache in sync with the database.
+const rateLimiters = new Map<number, RateLimiterRedis>();
+
+//We can make this dynamic based on the subscription as well
+const RATE_LIMIT_DURATION_IN_MINUTES = 1;
+
+const getRateLimiter = (rateLimit: number): RateLimiterRedis => {
+    if (!rateLimiters.has(rateLimit)) {
+        const opts: IRateLimiterStoreOptions = {
+            storeClient: redis,
+            points: rateLimit, // Points represent the maximum number of requests allowed within the set duration.
+            duration: RATE_LIMIT_DURATION_IN_MINUTES * 60, // Converts minutes to seconds for the duration.
+        };
+        rateLimiters.set(rateLimit, new RateLimiterRedis(opts));
+    }
+    return rateLimiters.get(rateLimit)!;
+};
+
+const rateLimitMiddleware = () => async (req: Request, res: Response, next: Function) => {
+    if (skipRateLimitRoutes(req)) next();
+    try {
+        const { 'x-revert-t-id': tenantId } = req.headers;
+        const { subscription, id: accountId } = res.locals.account; // Subscription details are retrieved from response locals set earlier in the revertAuthMiddleware.
+        const rateLimit = subscription?.rate_limit ?? config.DEFAULT_RATE_LIMIT_DEVELOPER_PLAN; //incase subscription undefined, we will use the default rate limit this is to make sure backward compatibility as currently some accounts might not have subscription attached to them. We can remove the optional chaining and nullish coalescing once we are sure that all accounts have subscription attached to them
```

Comment:
@shadab14meb346 could you update the fallback such that it takes the default value of 1 here as per `RATE_LIMIT_DURATION_IN_MINUTES` in case it is not present in `.env`